lib/puppet/defaults.rb

```diff
@@ -1517,8 +1517,7 @@ def self.default_diffargs
     },
     :lastrunreport =>  {
       :default  => "$statedir/last_run_report.yaml",
-      :type     => :file,
-      :mode     => "0640",
```

Comment:
I'm suspicious of losing this mode enforcement. Also, I'd like to update the description to be clear about how the path is used.